src/v2/providers/alerts/performance.ts

```diff
@@ -0,0 +1,165 @@
+  // Scope the function to trigger on a specific application.
+  appId?: string;
+}
+
+/**
+ * Declares a function that can handle receiving performance threshold alerts.
+ * @param handler - Event handler which is run every time a threshold alert is received.
+ * @returns A function that you can export and deploy.
+ */
+export function onThresholdAlertPublished(
+  handler: (event: PerformanceEvent<ThresholdAlertPayload>) => any | Promise<any>
+): CloudFunction<PerformanceEvent<ThresholdAlertPayload>>;
+
+/**
+ * Declares a function that can handle receiving performance threshold alerts.
+ * @param appId - A specific application the handler will trigger on.
+ * @param handler - Event handler which is run every time a threshold alert is received.
+ * @returns A function that you can export and deploy.
+ */
+export function onThresholdAlertPublished(
+  appId: string,
+  handler: (event: PerformanceEvent<ThresholdAlertPayload>) => any | Promise<any>
+): CloudFunction<PerformanceEvent<ThresholdAlertPayload>>;
+
+/**
+ * Declares a function that can handle receiving performance threshold alerts.
+ * @param opts - Options that can be set on the function.
+ * @param handler - Event handler which is run every time a threshold alert is received.
+ * @returns A function that you can export and deploy.
+ */
+export function onThresholdAlertPublished(
+  opts: PerformanceOptions,
+  handler: (event: PerformanceEvent<ThresholdAlertPayload>) => any | Promise<any>
+): CloudFunction<PerformanceEvent<ThresholdAlertPayload>>;
+
+/**
+ * Declares a function that can handle receiving performance threshold alerts.
+ * @param appIdOrOptsOrHandler - A specific application, options, or an event-handling function.
+ * @param handler - Event handler which is run every time a threshold alert is received.
+ * @returns A function that you can export and deploy.
+ */
+export function onThresholdAlertPublished(
+  appIdOrOptsOrHandler:
+    | string
+    | PerformanceOptions
+    | ((event: PerformanceEvent<ThresholdAlertPayload>) => any | Promise<any>),
+  handler?: (event: PerformanceEvent<ThresholdAlertPayload>) => any | Promise<any>
+): CloudFunction<PerformanceEvent<ThresholdAlertPayload>> {
+  if (typeof appIdOrOptsOrHandler === "function") {
+    handler = appIdOrOptsOrHandler as (
+      event: PerformanceEvent<ThresholdAlertPayload>
+    ) => any | Promise<any>;
+    appIdOrOptsOrHandler = {};
+  }
+
+  const [opts, appId] = getOptsAndApp(appIdOrOptsOrHandler);
+
+  const func = (raw: CloudEvent<unknown>) => {
+    return wrapTraceContext(handler)(raw as PerformanceEvent<ThresholdAlertPayload>);
+  };
+
+  func.run = handler;
+  func.__endpoint = getEndpointAnnotation(opts, thresholdAlert, appId);
+
+  return func;
+}
+
+/**
+ * Helper function to parse the function opts and appId.
+ * @internal
+ */
+export function getOptsAndApp(
+  appIdOrOpts: string | PerformanceOptions
+): [EventHandlerOptions, string | undefined] {
```

Comment:
Not really a nit or complaint, but curious why returning populated array instead of an object containing the fields?

```
[EventHandlerOptions, string | undefined]
vs
{opts: EventHandlerOptions, appId?: string}
```

Reply:
looks good, thanks for the improvement!

Reply:
Just b/c it was only a couple fields, I did a similar thing for the other alert triggers